File: java/src/priorityQueue/DisJointSetArray.java
package priorityQueue;

import java.util.Arrays;

public class DisJointSetArray {
    private int[] smallest = new int[9];

    private int size = -1;

    private int maxSize = 9;

    public String toString() {
        StringBuffer sb = new StringBuffer();
        for (int i = 0; i <= this.size; i++) {
            sb.append(this.smallest[i]);
            sb.append(",");
        }

        return sb.toString();
    }

    public void makeSet(int p) {
        if (this.size == this.maxSize - 1)
            throw new IndexOutOfBoundsException();

        this.smallest[++this.size] = p;
    }

    public int find(int i) {
        if (i < 0 || i > this.size)
            throw new IndexOutOfBoundsException();

        return this.smallest[i];
    }

    public void union(int i, int j) {
        int i_id = this.find(i);
        int j_id = this.find(j);

        if (i_id == j_id)
            return;

        int m = Math.min(i_id, j_id);

        for (int idx = 0; idx <= this.size; idx++) {
            if (this.smallest[idx] == i_id || this.smallest[idx] == j_id) {
                this.smallest[idx] = m;
            }
        }
    }
}

File: python/priorityQueue/disJointSetArray.py
class DisJointSetArray:
    def __init__(self):
        self.smallest = [None] * 9
        self.size = -1
        self.maxSize = 9

    def toString(self):
        return ','.join([str(self.smallest[i]) for i in range(self.size + 1)])

    def makeSet(self, p):
        if self.size == self.maxSize - 1:
            raise IndexError
        self.size += 1
        self.smallest[self.size] = p

    def find(self, i):
        if i < 0 or i > self.size:
            raise IndexError
        return self.smallest[i]

    def union(self, i, j):
        i_id = self.find(i)
        j_id = self.find(j)

        if i_id == j_id:
            return

        m = min(i_id, j_id)
        
        for idx in range(self.size + 1):
            if self.smallest[idx] in [i_id, j_id]:
                self.smallest[idx] = m
